viadot/tasks/azure_sql.py

```diff
@@ -264,3 +267,72 @@ def run(
 
         self.logger.info(f"Successfully ran the query.")
         return result
+
+
+class CheckColumnOrder(Task):
+    """
+    Task for checking the order of columns in the loaded DF and in the SQL table into which the data from DF will be loaded.
+    If order is different then DF columns are reordered according to the columns of the SQL table.
+    """
+
+    def __init__(
+        self,
+        table: str = None,
+        if_exists: Literal["fail", "replace", "append", "delete"] = "replace",
+        df: pd.DataFrame = None,
+        credentials_secret: str = None,
+        vault_name: str = None,
+        *args,
+        **kwargs,
+    ):
+        self.credentials_secret = credentials_secret
+        self.vault_name = vault_name
+
+        super().__init__(name="run_check_column_order", *args, **kwargs)
+
+    def df_change_order(
+        self, df: pd.DataFrame = None, sql_column_list: List[str] = None
+    ):
+        df_column_list = list(df.columns)
+        if set(df_column_list) == set(sql_column_list):
+            df_column_list = sql_column_list
```

Comment:
This line doesn't really change anything, as `df_column_list` is not used anywhere. Was it meant to be used later in `df_changed`?

In line 331 we are checking the equality of DataFrame columns and SQL columns. Is it really necessary to check for it again?

Reply:
A new version of the file was uploaded.
Line 334 is a comparison of two lists that are identical (whether the order is the same).
Line 297  is a comparison of the number of columns if they are the same (can be in a different order) - if so - the order is changed if not - an exception is thrown.
Make sense?

Reply:
Yup, I see. Looks fine to me 😄